### PR TITLE
fix: refactor Files metadata to generic resource metadata

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/data/ResourceMetadataCollection.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class ResourceMetadataCollection extends MetadataBase {
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/data/ResourceMetadataCollection.java]: I would use it be *Folder to match with NodeType enum.

File: src/main/java/com/epam/aidial/core/data/ResourceType.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.epam.aidial.core.data;
+
+import lombok.Getter;
+
+@Getter
+public enum ResourceType {
+    FILE("files");
+
+    private final String resourceGroup;
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/data/ResourceType.java]: nit: folder? or just group.

File: src/main/java/com/epam/aidial/core/controller/FileMetadataController.java [hunk 3]
```diff
@@ -21,7 +21,7 @@ protected Future<?> handle(ResourceDescription resource) {
         BlobStorage storage = proxy.getStorage();
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:21]: Are you going to change metadata API in this pr?

Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:21]: I don't have other resources to deal with right now. Let's implement is later

Comment (artsiomkorzun) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:21]: The frontend will be able to migrate on these change as soon as possible.

Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/controller/FileMetadataController.java:21]: I think I misunderstood your point. We should fix metadata routing in this PR

**Review by artsiomkorzun — APPROVED** [review]

**Pull request merged**